webgui/scripts/codecompass/view/component/Text.js

```diff
@@ -326,18 +327,8 @@ function (declare, domClass, dom, style, query, topic, ContentPane, Dialog,
      * file ID.
      */
     loadFile : function (file) {
-      if (!(file instanceof FileInfo)) {
-        // The same file shouldn't be loaded twice after each other.
-        if (this._fileInfo && file === this._fileInfo.id)
-          return;
-
-        this._fileInfo = model.project.getFileInfo(file);
-      } else {
-        this._fileInfo = file;
-      }
-
-      this.set('content', model.project.getFileContent(this._fileInfo.id));
-      this.set('header', this._fileInfo);
+      this.set('content', urlHandler.getFileContent());
+      this.set('header', urlHandler.getFileInfo());
```

Comment:
This `loadFile()` function shouldn't rely on the URL. Any plugin can call this `loadFile()` explicitly, rather those should check if the cached file can be reused when calling it.